perch/organisation.py

```diff
@@ -434,6 +433,9 @@ def authorized(self, requested_access, resource):
         :param resource: a Resource or SubResource with "permissions" attribute
         :returns: True if has access, False otherwise
         """
+        if {self.state, resource.state} != {State.approved.value}:
```

Comment:
OK. `authorized` will need to be changed into a `coroutine` in case the parent needs to be fetched